scipy/stats/stats.py

```diff
@@ -5819,7 +5689,185 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
+    or 'greater'. The KS test is only valid for continuous distributions.
+
+    Parameters
+    ----------
+    x : array_like
+        a 1-D array of observations of random variables.
+    cdf : callable
+        callable used to calculate the cdf.
+    args : tuple, sequence, optional
+        Distribution parameters, used with `cdf`.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': one-sided, see explanation in Notes
+          * 'greater': one-sided, see explanation in Notes
+    mode : {'auto', 'exact', 'approx', 'asymp'}, optional
+        Defines the distribution used for calculating the p-value.
+        The following options are available (default is 'auto'):
+
+          * 'auto' : selects one of the other options.
+          * 'exact' : uses the exact distribution of test statistic.
+          * 'approx' : approximates the two-sided probability with twice the one-sided probability
+          * 'asymp': uses asymptotic distribution of test statistic
+
+    Returns
+    -------
+    statistic : float
+        KS test statistic, either D, D+ or D-.
+    pvalue :  float
+        One-tailed or two-tailed p-value.
+
+    See Also
+    --------
+    ks_2samp, kstest
+
+    Notes
+    -----
+    In the one-sided test, the alternative is that the empirical
+    cumulative distribution function of the random variable is "less"
+    or "greater" than the cumulative distribution function G(x) of the
+    hypothesis, ``F(x)<=G(x)``, resp. ``F(x)>=G(x)``.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    >>> x = np.linspace(-15, 15, 9)
+    >>> stats.kstest(x, 'norm')
+    (0.44435602715924361, 0.038850142705171065)
+
+    >>> np.random.seed(987654321) # set random seed to get the same result
+    >>> stats.kstest('norm', False, N=100)
+    (0.058352892479417884, 0.8653960860778898)
+
+    The above lines are equivalent to:
+
+    >>> np.random.seed(987654321)
+    >>> stats.kstest(stats.norm.rvs(size=100), 'norm')
+    (0.058352892479417884, 0.8653960860778898)
+
+    *Test against one-sided alternative hypothesis*
+
+    Shift distribution to larger values, so that ``cdf_dgp(x) < norm.cdf(x)``:
+
+    >>> np.random.seed(987654321)
+    >>> x = stats.norm.rvs(loc=0.2, size=100)
+    >>> stats.kstest(x,'norm', alternative = 'less')
+    (0.12464329735846891, 0.040989164077641749)
+
+    Reject equal distribution against alternative hypothesis: less
+
+    >>> stats.kstest(x,'norm', alternative = 'greater')
+    (0.0072115233216311081, 0.98531158590396395)
+
+    Don't reject equal distribution against alternative hypothesis: greater
+
+    >>> stats.kstest(x,'norm', mode='asymp')
+    (0.12464329735846891, 0.08944488871182088)
+
+    *Testing t distributed random variables against normal distribution*
```

Comment:
the example below is much easier to follow than the one above that has less comments on the idea behind the tests

Reply:
I believe that the docstring for this function was copied verbatim from the docstriong for the existing `stats.kstest`. If you have replacement doc examples you prefer now I' be happy to use them! If not, I'd like to make those changes in a different PR, as it will likely affect multiple functions.